docs/rules/new-for-builtins.md

```diff
@@ -0,0 +1,70 @@
+# Enforce the use of `new` for all builtins
+
+They work the same, but `new` should be preferred for consistency with other constructors.
+
+Enforces the use of `new` for following builtins.
+
+- `Array`
+- `ArrayBuffer`
+- `DataView`
+- `Date`
+- `Error`
+- `Float32Array`
+- `Float64Array`
+- `Function`
+- `Int8Array`
+- `Int16Array`
+- `Int32Array`
+- `Map`
+- `WeakMap`
+- `Set`
+- `WeakSet`
+- `Promise`
+- `RegExp`
+- `Symbol`
+- `Uint8Array`
+- `Uint16Array`
+- `Uint32Array`
+- `Uint8ClampedArray`
```

Comment:
Just realized, we need to include `Object` here instead of the below. `Object` and `new Object` both wraps the object and doesn't coerce. And the `no-new-object` ESLint rule only catches `new Object()`, not `Object`.